core/src/main/java/org/kiwix/kiwixmobile/core/downloader/ChunkUtils.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.downloader
+
+import org.kiwix.kiwixmobile.core.utils.StorageUtils.getFileNameFromUrl
+import java.util.ArrayList
+import java.util.Locale
+
+object ChunkUtils {
+  private const val ALPHABET = "abcdefghijklmnopqrstuvwxyz"
+  private const val ZIM_EXTENSION = ".zim"
+  const val PART = ".part.part"
+  const val CHUNK_SIZE = 1024L * 1024L * 1024L * 2L
+
+  fun getChunks(
+    url: String?,
+    contentLength: Long,
+    notificationID: Int
+  ): List<Chunk> {
+    val fileCount = getZimChunkFileCount(contentLength)
+    val filename = getFileNameFromUrl(url)
+    val fileNames = getZimChunkFileNames(filename, fileCount)
+    return generateChunks(contentLength, url, fileNames, notificationID)
+  }
+
+  private fun generateChunks(
+    contentLength: Long,
+    url: String?,
+    fileNames: Array<String?>,
+    notificationID: Int
+  ): List<Chunk> {
+    val chunks: MutableList<Chunk> = ArrayList()
+    var currentRange: Long = 0
+    var minLen: Long
+    for (zim in fileNames) {
+      minLen = minOf(
+        contentLength, (currentRange + CHUNK_SIZE)
+      )
+      val range: String = if (currentRange + CHUNK_SIZE >= contentLength) {
+        String.format(Locale.US, "%d-", currentRange)
```

Comment:
We should be able to replace this with kotlin string interpolation, should make it much easier to read

Reply:
Changes done. Please review

Reply:
Not using kotlin string interpolation